stl/inc/chrono

```diff
@@ -5428,113 +5466,67 @@ struct formatter<_CHRONO day, _CharT> {
-        _Conv_spec                              = _Target_conv_spec;
-        // only 'd' and 'e' acceptable types for day
-        if (_Target_conv_spec._Type != 'd' && _Target_conv_spec._Type != 'e') {
-            _THROW(format_error("Invalid type for type day."));
-        }
-
-        if (_Target_conv_spec._Modifier != '\0' && _Target_conv_spec._Modifier != 'O') {
-            _THROW(format_error("Invalid modifier for type day."));
-        }
-
-        _Target_conv_spec_iter =
-            _STD find_if(_Target_conv_spec_iter + 1, _List.end(), [](const auto& _Cs) { return _Cs._Type != '\0'; });
-
-        if (_Target_conv_spec_iter != _List.end()) {
-            _THROW(format_error("day requires one conversion-spec."));
+        for (const auto& _Spec : _List) {
+            if (_Spec._Type != '\0' && _Spec._Type != 'd' && _Spec._Type != 'e') {
+                _THROW(format_error("Invalid type for type day."));
+            }
+            if (_Spec._Modifier != '\0' && _Spec._Modifier != 'O') {
+                _THROW(format_error("Invalid modifier for type day."));
+            }
         }
 
-        _Stored_specs = _Specs;
         return _Res_iter;
     }
 
     template <class _FormatContext>
     typename _FormatContext::iterator format(const _CHRONO day& _Val, _FormatContext& _FormatCtx) {
         using _Ctx_iter = typename _FormatContext::iterator;
 
-        const basic_string<_CharT> _Invalid_msg = _STATICALLY_WIDEN(_CharT, " is not a valid day");
         basic_stringstream<_CharT> _Stream;
-        _Stream.imbue(_FormatCtx.locale());
-        basic_string<_CharT> _Str;
-        int _Num_char = 2;
-        if (_No_chrono_specs && !_Val.ok()) {
-            _Stream << static_cast<unsigned int>(_Val);
+
+        if (_No_chrono_specs) {
             if (static_cast<unsigned int>(_Val) < 10) { // _Val == 0
-                if (_Conv_spec._Type == 'd') {
-                    _Str = _CharT{'0'} + _Stream.str();
-                } else { // e
-                    _Str = _CharT{' '} + _Stream.str();
-                }
-            } else {
-                _Str = _Stream.str();
+                _Stream << _CharT{'0'};
+            }
+            _Stream << static_cast<unsigned int>(_Val);
+            if (!_Val.ok()) {
+                _Stream << _STATICALLY_WIDEN(_CharT, " is not a valid day");
             }
-            _Str = _Str + _Invalid_msg;
-            _Num_char += static_cast<int>(_Invalid_msg.length());
         } else {
+            _Stream.imbue(_FormatCtx.locale());
             tm _Time = {.tm_mday = static_cast<int>(static_cast<unsigned int>(_Val))};
-            _CharT _Fmt_str[4];
-            _Fmt_str[0]      = _CharT{'%'};
-            size_t _Next_idx = 1;
-            if (_Conv_spec._Modifier != '\0') {
-                _Fmt_str[_Next_idx++] = _Conv_spec._Modifier;
-            }
+            for (auto _Spec_it = _Specs._Chrono_specs_list.begin(); _Spec_it != _Specs._Chrono_specs_list.end();) {

```

Comment:
I was initially confused by this loop-containing-a-loop; I see what it's doing now, but I believe that it contains dead code and could be significantly simplified.

The loop assumes that we start with a conversion-spec (which will be enforced by @mnatsuhara) and prints it with `put_time` which is correct. But then it tests whether `if (_Spec_it->_Lit_char != _CharT{'\0'}) {` which is always false - nothing can be a conversion-spec and a literal-char simultaneously (we digest `%n` `%t` and `%%` immediately as literal-chars). The nested loop is correct but weird (it prints out literal-chars until we find the next conversion-spec if any, which re-establishes the invariant/assumption needed by the outer loop).

I believe that this would be significantly simpler if flattened into a single loop, that increments `++_Spec_it`. For each element, it would test the `_Type` (or the `_Lit_char`, either is fine) and decide whether to print it with `put_time` or as a literal character.

A flat loop would lack the assumption that a conversion-spec is first, but that's okay - we need enforcement elsewhere either way.